tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -57,6 +57,12 @@ private static function findTestFiles(): iterable
 			yield __DIR__ . '/data/explode-php80.php';
 		}
 
+		if (PHP_VERSION_ID < 80000) {
+			yield __DIR__ . '/data/getenv-php74.php';
+		} else {
+			yield __DIR__ . '/data/getenv-php80.php';
```

Comment:
This doesn't need to be in NodeScopeResolverTest. Move the files to nsrt/ and label them with proper `// lint` conditions.

Reply:
There is already similar conditions checks like above
```
if (PHP_VERSION_ID < 80000) {
			yield __DIR__ . '/data/explode-php74.php';
		} else {
			yield __DIR__ . '/data/explode-php80.php';
		}
```

The expectated type are different based on the PHP version, that's why I split the files.

The `// lint` condition is only used for the linter right ? Not for the tests

Reply:
Files in nsrt/ when testing type inference are filtered based on`// lint` as well.